polkadot/node/network/collator-protocol/src/validator_side/collation.rs

```diff
@@ -148,6 +148,9 @@ pub fn fetched_collation_sanity_check(
 		.prospective_candidate
 		.map_or(false, |pc| pc.candidate_hash() != fetched.hash())
 	{
+		// Note: it's important that we check for this and punish a collator that advertises a false
+		// candidate hash, because it can be misused to prioritise a specific collation, according
+		// to the fork choice rule in Prospective Parachains.
```

Comment:
The prioritization only happens when fully backed. So this seems unrelated?

Reply:
Even if the prioritisation happens when fully backed, it may be misused to prioritise a candidate. I'm not sure we check the candidate's hash against the provided candidate hash somewhere else (during validation).

We do this check here and in statement distribution and punish the reputation if misused.

Reply:
I guess my point is: Advertising a hash and delivering something else is malicious. Trying to manipulate the fork choice rule would be the least of my worries.

Reply:
Ok I get what you mean now. Fair point

Reply:
removed this comment